### ISSUE TITLE
Model parameter tuning in parallel

### ISSUE THREAD

**Issue by sumedhvdatar** [uber/horovod#437]
Using horovod can we perform hyperparameter tuning using Keras. For example I have 6 gpus and I want to provide different parameters for each model and run in parallel. Is there a way we can run the models in parallel and get the best model or probably the report of how each model performed with the specified hyperparameters in parallel.

**Comment by alsrgv** [comment]
@sumedhvdatar, it's actually better to use distributed 6-GPU training to get results for each experiment ~6x times faster, since in that case, next experiment can use prior experiment results to decide which parameters to use.  This is particularly useful for Bayesian optimization.  Hyperas/Hyperopt uses TPE algorithm described [here](https://arxiv.org/pdf/1209.5111.pdf), which will benefit from such prior knowledge.

**Comment by sumedhvdatar** [comment]
@alsrgv I ran two codes one with just hyperas and only 1 GPU and the second code, the one you sent with 2 gpus. I see that Horovod code is actually taking more time than running hyperas on 1 GPU. Is there an alternative to this?

**Comment by alsrgv** [comment]
@sumedhvdatar, did you install Horovod with NCCL?  I just did an experiment on P100 with NCCL, and I don't see such drastic regression.

* 1xP100 - 3m25.238s
* 2xP100 - 4m10.933s
* 4xP100 - 2m18.437s

Note, that 2xP100 is slower than 1xP100 because MNIST is such a small model that communication has quite significant cost compared to model computation.  You will see much better scaling with larger deep learning models.

**Comment by sumedhvdatar** [comment]
@alsrgv  thanks for verifying. Can you elaborate more on installing horovod with NCCL?

**Comment by sumedhvdatar** [comment]
@alsrgv  i am using Tesla K80. Let me check the documentation and run the experiment. Many thanks!

**Comment by sumedhvdatar** [comment]
@alsrgv  I ran horovod with hyperas on NC-6 and NC-24. From my experiment I found out that Horovod with hyperas ran faster on NC6 than NC-24. Looks like data is parallel on horovod but hyperas is more sequential and running the same thing on all gpus again. Can you help me here?

**Comment by sumedhvdatar** [comment]
@alsrgv  I feel that logs from Keras with horovod is adding to the additional time in NC24. I noticed that in NC24 , during training it was printing four times, the epoch and progress bar. Making verbose = 0 may help in reducing training time. I ll give you an update after running without logs.

Also currently when I ran the code NC12 was faster than NC6 but NC24 and NC6 was almost the same. Major factor could be logs.
